src/H5Dchunk.c

```diff
@@ -297,9 +296,9 @@ static herr_t   H5D__create_piece_file_map_all(H5D_dset_io_info_t *di, H5D_io_in
 static herr_t   H5D__create_piece_file_map_hyper(H5D_dset_io_info_t *di, H5D_io_info_t *io_info);
 static herr_t   H5D__create_piece_mem_map_1d(const H5D_dset_io_info_t *di);
 static herr_t   H5D__create_piece_mem_map_hyper(const H5D_dset_io_info_t *di);
-static herr_t   H5D__piece_file_cb(void *elem, const H5T_t *type, unsigned ndims, const hsize_t *coords,
```

Comment:
There a lot of `const`s that had to be removed in this PR, mostly because the `H5T_t *` for the datatypes aren't being hidden behind an `hid_t` anymore and are used directly, where they end up getting passed to either `H5T_close` or `H5T_convert`, both of which would require casting const away. If we're able to mark the src and dst datatype parameters to `H5T_convert` as `const`, a good chunk of these can probably come back, but I wasn't sure if that was reasonable to do. Usage of `H5T_close` is more problematic.

Reply:
I don't think that those H5T_t's to H5T_convert can be const, since their state gets updated for certain conversions.  (I'm thinking vlen & ref conversions)

Reply:
No problem removing const here...

Reply:
At least from a quick scan it doesn't look like the pointers themselves get modified, but parts of the pointed to struct, like the file pointer for vlen and ref types, might get modified. Though, depending on what's actually modified we could get away with it, as long as the direct contents of the pointed to H5T_t aren't modified. Would have to dig deeper to see if there would be problems, but could be interesting.